### PR TITLE
Cleanup the `Tracing` class and move setup as instance method of `TracingConfig`

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Followup of #900

## Proposed Changes

- :broom: Cleanup the `Tracing` class and move setup as instance method of `TracingConfig`
- :wastebasket: Removed `TracingConfig.Builder` as it was used only internally

### CODE REVIEW

**Review by matzew — APPROVED** [review]
from a refactoring perspective this 
/lgtm

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingConfig.java [hunk 1]
```diff
@@ -39,70 +69,124 @@ private TracingConfig(final Backend backend, final String url, final float sampl
...
-    if (!Files.exists(backendPath)) {
-      return TracingConfig.builder()
-        .setBackend(Backend.UNKNOWN)
-        .createTracingConfig();
+  public OpenTelemetrySdk setup() {
+    logger.info(
+      "Registering tracing configurations {} {} {} {}",
+      keyValue("backend", getBackend()),
+      keyValue("sampleRate", getSamplingRate()),
+      keyValue("url", getUrl()),
```
Comment (pierDipi) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingConfig.java:84]: `url` might contain sensitive data, keys or passwords of a monitoring provider, even if config-tracing by default is a CM, it's potentially possible to convert and use a secret but we should print it here.

**Review by pierDipi — COMMENTED** [review]
LGTM

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingConfig.java (hunk 1 above)
Comment (slinkydeveloper) [reply data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/tracing/TracingConfig.java:84]: I don't recall this was there before this cleanup or not TBH, but I'll pr it

**Pull request merged**